basic-steps/CORE-STEPS.md

```diff
@@ -72,8 +72,28 @@ node {
 though this would not automatically adjust the message according to the status of _previous_ builds as the standard mail notifier does.
 That would be possible only via [JENKINS-26834](https://issues.jenkins-ci.org/browse/JENKINS-26834).
 
+# Build wrappers
+
+The `wrap` step may be used to run a build wrapper defined originally for freestyle projects.
+In a workflow, any block of code (inside `node`) may be wrapped in this way, not necessarily the whole build.
+
+For example, the Xvnc plugin allows a headless build server to run GUI tests by allocating an in-memory-only X11 display.
+To use this plugin from a workflow, assuming a version with the appropriate update:
+
+```groovy
+node('linux') {
+  wrap([$class: 'Xvnc', useXauthority: true]) {
```

Comment:
Good question. It will not. Making it do so would require `CoreWrapperStep` to define a `PickleFactory` for a `SimpleBuildWrapper` (which would also need to be `Serializable`), and then there would need to be some API allowing a `Pickle` to differentiate between being deserialized as part of a Jenkins restart vs. being deserialized in another `Run` as part of a checkpoint. Even that is probably not enough, because simply calling `setUp` again will not work in all cases; say the original Xvnc server set `DISPLAY=:11` and the new one sets `DISPLAY=:12` yet some code in the block is somehow caching the environment.

And adding subtle logic like this is out of the question without good test coverage, which is hard since the checkpoints are not available in OSS, so the tests would need to be in _that_ plugin, and branched to match this one.

Anyway our [documented recommendation](http://jenkins-enterprise.cloudbees.com/docs/user-guide-docs/workflow-sect-checkpoint.html#workflow-sect-checkpoint-files) for checkpoints is that they not be used inside a `node` block to begin with. There are too many things that can go wrong; you would need to get the exact same slave back, with an untouched workspace. And if checkpoints live at “top level” (i.e., in a scope not holding locks on any external resources), then they cannot be inside `wrap` either.